Mathlib/GroupTheory/Exponent.lean

```diff
@@ -178,25 +189,46 @@ theorem order_dvd_exponent (g : G) : orderOf g ∣ exponent G :=
 #align monoid.order_dvd_exponent Monoid.order_dvd_exponent
 #align add_monoid.add_order_dvd_exponent AddMonoid.addOrder_dvd_exponent
 
+@[to_additive]
+theorem orderOf_le_exponent (h : ExponentExists G) (g : G) : orderOf g ≤ exponent G :=
+  Nat.le_of_dvd h.exponent_pos (order_dvd_exponent g)
+
 variable (G)
 
 @[to_additive]
-theorem exponent_dvd_of_forall_pow_eq_one (G) [Monoid G] (n : ℕ) (hG : ∀ g : G, g ^ n = 1) :
-    exponent G ∣ n := by
+theorem exponent_dvd_iff_forall_pow_eq_one {n : ℕ} : exponent G ∣ n ↔ ∀ g : G, g ^ n = 1 := by
   rcases n.eq_zero_or_pos with (rfl | hpos)
-  · exact dvd_zero _
-  apply Nat.dvd_of_mod_eq_zero
-  by_contra h
-  have h₁ := Nat.pos_of_ne_zero h
-  have h₂ : n % exponent G < exponent G := Nat.mod_lt _ (exponent_pos_of_exists n hpos hG)
-  have h₃ : exponent G ≤ n % exponent G := by
-    apply exponent_min' _ h₁
-    simp_rw [← pow_eq_mod_exponent]
-    exact hG
-  linarith
+  · simp
+  constructor
+  · intro h g
+    rw [Nat.dvd_iff_mod_eq_zero] at h
+    rw [pow_eq_mod_exponent, h, pow_zero]
+  · intro hG
+    by_contra h
+    rw [Nat.dvd_iff_mod_eq_zero, ← Ne, ← pos_iff_ne_zero] at h
+    have h₂ : n % exponent G < exponent G := Nat.mod_lt _ (exponent_pos_of_exists n hpos hG)
+    have h₃ : exponent G ≤ n % exponent G := by
+      apply exponent_min' _ h
+      simp_rw [← pow_eq_mod_exponent]
+      exact hG
+    exact h₂.not_le h₃
+
+@[to_additive (attr := deprecated)]
+theorem exponent_dvd_of_forall_pow_eq_one (n : ℕ) (hG : ∀ g : G, g ^ n = 1) :
+    exponent G ∣ n :=
```

Comment:
I think it would be nice to keep this around without the deprecation
```suggestion
@[to_additive]
theorem exponent_dvd_of_forall_pow_eq_one (n : ℕ) (hG : ∀ g : G, g ^ n = 1) :
    exponent G ∣ n :=
```

Reply:
Can we make it an alias then?